src/goto-instrument/reachability_slicer.cpp

```diff
@@ -86,13 +89,12 @@ void reachability_slicert::fixedpoint_to_assertions(
       {
         // This is an end-of-function -> successor-of-callsite edge.
         // Queue both the caller and the end of the callee.
-        INVARIANT(
-          std::prev(node.PC)->is_function_call(),
-          "all function return edges should point to the successor of a "
-          "FUNCTION_CALL instruction");
-        stack.emplace_back(edge.first, true);
-        stack.emplace_back(
-          cfg.entry_map[std::prev(node.PC)], caller_is_known);
+        if(std::prev(node.PC)->is_function_call())
```

Comment:
Comment about when this might be false?

Reply:
It really shouldn't happen, but something broke in the slicer during multiple calls and the cfg-predecessor of a successor-of-callsite was not a function call. Now that a new slicer is created it does not happen, so I replaced it back with the original invariant.